app/src/main/java/org/mozilla/fenix/compose/HorizontalFadingEdge.kt

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+
+/**
+ * Displays the [content] with the right edge fading.
+ *
+ * @param modifier [Modifier] for the container.
+ * @param fadeWidth Length of the fading edge.
+ * @param backgroundColor Color of the background shown under the content.
+ * @param isContentRtl Whether or not the content should be displayed Right to Left
+ * @param content The content whose right edge must be faded.
+ */
+@Composable
+fun HorizontalFadingEdge(
+    modifier: Modifier = Modifier,
+    fadeWidth: Dp = 25.dp,
+    backgroundColor: Color = Color.Transparent,
+    isContentRtl: Boolean = false,
```

Comment:
you may be able to remove this parameter and use `LocalLayoutDirection.current == LayoutDirection.Rtl` instead.

Reply:
This is what I used initially to decide the direction of the tab title. However, compared to the current implementation (`TextView` with `fadingEdgeLength`), the direction of the text and the fading edge would be this way imposed by the device locale. In the current implementation the direction of the title text is based on the content of the title. Here are some screenshots I obtained using `LocalLayoutDirection.current == LayoutDirection.Rtl `:

|Current implementation | Compose version |
| --- | --- |
| ![Screenshot_20220720_154051](https://user-images.githubusercontent.com/35462038/179985360-40636d60-b187-46d8-8ed2-65f10ffcf928.png)|![changed](https://user-images.githubusercontent.com/35462038/179985404-3695ca5e-16fe-4f6e-ae71-0a3785f64465.png)|
|Current implementation with RTL layout | Compose version with RTL layout|
|![Screenshot_20220720_154208](https://user-images.githubusercontent.com/35462038/179985644-be15b992-153e-4134-8ada-2ac9cc1ba405.png)|![Screenshot_20220720_154358](https://user-images.githubusercontent.com/35462038/179985683-304e6770-0b6d-4f93-b40d-5cf79c46e5a8.png)|

Reply:
Oh dang, interesting, ok. Good catch on this then.